include/rocksdb/trace_record_result.h

```diff
@@ -0,0 +1,145 @@
+      return Status::NotSupported(
+          "MultiValuesTraceExecutionResult is not supported.");
+    }
+  };
+
+  virtual Status Accept(Handler* handler) = 0;
+
+ private:
+  TraceType trace_type_;
+};
+
+// Base class for the execution results types of trace records.
+class TraceExecutionResult : public TraceRecordResult {
+ public:
+  TraceExecutionResult(uint64_t latency, TraceType trace_type);
+
+  virtual ~TraceExecutionResult() override;
+
+  // Request latency in nanoseconds.
+  virtual uint64_t GetLatency() const;
+
+ private:
+  uint64_t latency_;
+};
+
+// Operation that only returns a Status.
+// Example operations: DB::Write(), Iterator::Seek() and
+// Iterator::SeekForPrev().
+class StatusOnlyTraceExecutionResult : public TraceExecutionResult {
+ public:
+  StatusOnlyTraceExecutionResult(const Status& status, uint64_t latency,
+                                 TraceType trace_type);
+  virtual ~StatusOnlyTraceExecutionResult() override;
+
+  // Return value of DB::Write(), etc.
+  virtual Status GetStatus() const;
+
+  virtual Status Accept(Handler* handler) override;
+
+ private:
+  Status status_;
+};
+
+// Operation that returns a Status and a value.
+// Example operation: DB::Get()
+class SingleValueTraceExecutionResult : public TraceExecutionResult {
+ public:
+  SingleValueTraceExecutionResult(const Status& status,
+                                  const std::string& value, uint64_t latency,
+                                  TraceType trace_type);
+
+  SingleValueTraceExecutionResult(const Status& status, std::string&& value,
+                                  uint64_t latency, TraceType trace_type);
+
+  virtual ~SingleValueTraceExecutionResult() override;
+
+  // Return status of DB::Get(), etc.
+  virtual Status GetStatus() const;
+
+  // Value for the searched key.
+  virtual std::string GetValue() const;
+
+  virtual Status Accept(Handler* handler) override;
+
+ private:
+  Status status_;
+  std::string value_;
+};
+
+// Operation that returns a vector of values and status.
+// Example operation: DB::MultiGet()
+class MultiValuesTraceExecutionResult : public TraceExecutionResult {
+ public:
+  MultiValuesTraceExecutionResult(const std::vector<Status>& multi_status,
+                                  const std::vector<std::string>& values,
+                                  uint64_t latency, TraceType trace_type);
+
+  MultiValuesTraceExecutionResult(std::vector<Status>&& multi_status,
+                                  std::vector<std::string>&& values,
+                                  uint64_t latency, TraceType trace_type);
+
+  virtual ~MultiValuesTraceExecutionResult() override;
+
+  // Return status of DB::MultiGet(), etc.
+  virtual std::vector<Status> GetMultiStatus() const;
+
+  // Values for the searched keys.
+  virtual std::vector<std::string> GetValues() const;
+
+  virtual Status Accept(Handler* handler) override;
+
+ private:
+  std::vector<Status> multi_status_;
+  std::vector<std::string> values_;
```

Comment:
I guess you want this to be able to present bytes/sec. I think we are OK starting with QPS, though.

Reply:
This might be used for bytes/sec, but it's not the major goal. Currently you can get the TraceRecord execution status from `Replayer::Next` and `Replayer::Execute`, but you have no way to obtain the actual DB execution status from these 2 APIs. Also you cannot get any execution status from `Replayer::Replay`. 

The `TraceRecordResult` API set provides an interface to query the execution result from `TraceRecord::Handler`. If it is the execution handler, then you can get the DB calls returned status and/or values.

If we use Handlers for TraceAnalyzer as well, `TraceRecordResult` may be used for parsing or outputting to stdout or file.